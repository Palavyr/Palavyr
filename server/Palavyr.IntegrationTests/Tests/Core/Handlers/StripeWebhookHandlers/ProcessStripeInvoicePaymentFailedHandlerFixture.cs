using System.Threading.Tasks;
using Autofac;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Logging;
using Palavyr.Core.Models.Accounts.Schemas;
using Palavyr.Core.Services.EmailService.ResponseEmailTools;
using Palavyr.Core.Services.StripeServices.StripeWebhookHandlers.PaymentFailed;
using Palavyr.IntegrationTests.AppFactory.AutofacWebApplicationFactory;
using Palavyr.IntegrationTests.DataCreators.StripeBuilders;
using Palavyr.IntegrationTests.Tests.Mocks;
using Test.Common.ApprovalTests;
using Xunit;
using Xunit.Abstractions;

namespace Palavyr.IntegrationTests.Tests.Core.Handlers.StripeWebhookHandlers
{
    public class ProcessStripeInvoicePaymentFailedHandlerFixture : StripeServiceFixtureBase
    {
        [Fact]
        public async Task HandlesEvent()
        {
            var invoice = await this.CreateStripeInvoiceBuilder().Build();
            var @event = new InvoicePaymentFailedEvent(invoice);
            var emailer = Container.GetService<ISesEmail>();
            var logger = Container.GetService<ILogger<ProcessStripeInvoicePaymentFailedHandler>>();
            var handler = new ProcessStripeInvoicePaymentFailedHandler(logger, ResolveStore<Account>(), emailer);

            await handler.Handle(@event, CancellationToken);

            var ses = (IGetEmailSent)Container.GetService<ISesEmail>();
 
            this.PalavyrAssent(ses.GetSentHtml());
        }

        public override ContainerBuilder CustomizeContainer(ContainerBuilder builder)
        {
            builder.RegisterType<MockSeSEmail>().As<ISesEmail>().SingleInstance();
            return base.CustomizeContainer(builder);
        }

        public ProcessStripeInvoicePaymentFailedHandlerFixture(ITestOutputHelper testOutputHelper, IntegrationTestAutofacWebApplicationFactory factory) : base(testOutputHelper, factory)
        {
        }
    }
}