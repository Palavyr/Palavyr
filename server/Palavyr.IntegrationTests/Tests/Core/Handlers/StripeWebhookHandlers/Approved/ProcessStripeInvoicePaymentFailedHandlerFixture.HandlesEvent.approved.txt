
<!DOCTYPE html>
<html lang='en'>

<head>
    <meta charset='UTF-8'>
    <meta name='viewport' content='width=device-width, initial-scale=1.0'>
    <title>Palavyr Email Confirmation</title>
</head>

<body style=' background: white; display: flex; justify-content: center; width: 100%'>
    <div id='card' style='background: #F2F2F2; width: 75%;'>
        <section
            style='color: white; text-align: center; border-radius: 7px; background-color: #507FE0; padding-top: 2rem; padding-left: 2rem; padding-right: 2rem; margin-top: 2rem; margin-left: 2rem; margin-right: 2rem;'
            id='header'>
            <div style='text-align: center; height: 60%;'><img style='border-radius: 16px;'
                    src='https://palavyr-public-assets.s3.amazonaws.com/LogoMedium.png' /></div>
            <br />
        </section>
        <section
            style='border-radius: 5%;  text-align: left; margin-left: 25%; padding-left: 2rem; margin-right: 25%; background-color: white; padding-top: 2rem; margin-top: 2rem;'
            id='body'>
            <h2>
                Oh No! Your latest subscription payment has failed!
            </h2>
            <p>
                Don't worry, we aren't going to cancel your subscription yet or anything :D
            </p>
            <br />
            <p>
                We typically allow a buffer period to allow you to update your payment method and try again. We'll continue trying to charge your current payment method just in case there was a temporary problem.
            </p>
            <p>
                Your subscription will lapse 27 days from now on Saturday, March 26, 2022
            </p>
            <p>
            <p>
                In the meantime, if you wish to continue using Palavyr, you can log into your account and navigate to your payment details using the 'Manage' tab in the sidebar navigation menu.
            </p>
            <p>
                If you have any questions or need any help, reach out to us at info.palavyr@gmail.com
            </p>
            <p>
                Sincerely,
                <br /><br />
                <strong>
                    The Palavyr Team
                </strong>
                <br />
                <br />
                <br />
            </p>
        </section>
    </div>
</body>

</html>
